src/core/ext/filters/client_channel/lb_policy/round_robin/round_robin.c

```diff
@@ -421,16 +425,15 @@ static grpc_connectivity_state update_lb_connectivity_status_locked(
                                 GRPC_CHANNEL_CONNECTING, GRPC_ERROR_NONE,
                                 "rr_connecting");
     return GRPC_CHANNEL_CONNECTING;
-  } else if (p->num_subchannels == 0) { /* 3) SHUTDOWN */
+  } else if (p->num_shutdown == p->num_subchannels) { /* 3) SHUTDOWN */
     grpc_connectivity_state_set(exec_ctx, &p->state_tracker,
-                                GRPC_CHANNEL_SHUTDOWN, GRPC_ERROR_REF(error),
```

Comment:
That was a leak. The ref is already incremented [here](https://github.com/dgquintas/grpc/blob/bc6bc090f471ac60c28a00152c637522e3bc72aa/src/core/ext/filters/client_channel/lb_policy/round_robin/round_robin.c#L471). This codepath wasn't hit prior to this PR.